std/experimental/allocator/package.d

```diff
@@ -1539,14 +1539,26 @@ exception if the copy operation throws.
 T[] makeArray(T, Allocator)(auto ref Allocator alloc, size_t length)
 {
     if (!length) return null;
+    static if (T.sizeof <= 1)
+    {
+        const nAlloc = length * T.sizeof;
+    }
+    else
+    {
+        import core.checkedint : mulu;
+        bool overflow;
+        const nAlloc = mulu(length, T.sizeof, overflow);
+        if (overflow) return null;
```

Comment:
Why not use `checkedint`?
It's what it was made for...

Reply:
e.g.

```d
scope(failure) return null;
return new int[(Checked!(size_t, Throw)(size_t.max) * size_t.max).get];
```

https://run.dlang.io/is/HiyHyD

Reply:
I feel that using a non-`@nogc` solution in `std.experimental.allocator` would defeat the purpose for many use cases.

Reply:
Yes, until there's a `checkedint` hook that just does an `assert(0)` on error it's better to just use the `core.checkedint` functions

Reply:
Because the use case here is to return `null` when an overflow is found, I don't think `Checked` is really usable here.